From 36bbdce1d62190f98bddf047ed9743444cfeb537 Mon Sep 17 00:00:00 2001
From: Scott Graham <sgraham@gmail.com>
Date: Sun, 12 Jan 2025 22:08:28 -0800
Subject: [PATCH 02/13] Fix warnings from clang about unprototyped vs (void)

e.g.

clang -O3 -std=c99 -g -Wall -Wextra -Wpedantic -c cfg.c -o cfg.o
cfg.c:4:7: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
    4 | newblk()
      |       ^
      |        void
1 warning generated.
---
 cfg.c   |  2 +-
 parse.c | 18 +++++++++---------
 rega.c  |  2 +-
 util.c  |  2 +-
 4 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/cfg.c b/cfg.c
index 8047c12..5db8c13 100644
--- a/cfg.c
+++ b/cfg.c
@@ -1,7 +1,7 @@
 #include "all.h"
 
 Blk *
-newblk()
+newblk(void)
 {
 	static Blk z;
 	Blk *b;
diff --git a/parse.c b/parse.c
index 7ab3ea5..406d13f 100644
--- a/parse.c
+++ b/parse.c
@@ -185,7 +185,7 @@ err(char *s, ...)
 }
 
 static void
-lexinit()
+lexinit(void)
 {
 	static int done;
 	int i;
@@ -207,7 +207,7 @@ lexinit()
 }
 
 static int64_t
-getint()
+getint(void)
 {
 	uint64_t n;
 	int c, m;
@@ -228,7 +228,7 @@ getint()
 }
 
 static int
-lex()
+lex(void)
 {
 	static char tok[NString];
 	int c, i, esc;
@@ -338,7 +338,7 @@ Alpha:
 }
 
 static int
-peek()
+peek(void)
 {
 	if (thead == Txxx)
 		thead = lex();
@@ -346,7 +346,7 @@ peek()
 }
 
 static int
-next()
+next(void)
 {
 	int t;
 
@@ -356,7 +356,7 @@ next()
 }
 
 static int
-nextnl()
+nextnl(void)
 {
 	int t;
 
@@ -421,7 +421,7 @@ tmpref(char *v)
 }
 
 static Ref
-parseref()
+parseref(void)
 {
 	Con c;
 
@@ -586,7 +586,7 @@ findblk(char *name)
 }
 
 static void
-closeblk()
+closeblk(void)
 {
 	idup(curb, insb, curi-insb);
 	blink = &curb->link;
@@ -1017,7 +1017,7 @@ parsefields(Field *fld, Typ *ty, int t)
 }
 
 static void
-parsetyp()
+parsetyp(void)
 {
 	Typ *ty;
 	int t, al;
diff --git a/rega.c b/rega.c
index ab98cd6..3be4c34 100644
--- a/rega.c
+++ b/rega.c
@@ -247,7 +247,7 @@ pmrec(enum PMStat *status, int i, int *k)
 }
 
 static void
-pmgen()
+pmgen(void)
 {
 	int i;
 	enum PMStat *status;
diff --git a/util.c b/util.c
index 3b5c09d..ad26f23 100644
--- a/util.c
+++ b/util.c
@@ -90,7 +90,7 @@ alloc(size_t n)
 }
 
 void
-freeall()
+freeall(void)
 {
 	void **pp;
 
-- 
2.49.0.vfs.0.4

